include/flatbuffers/flatbuffers.h

```diff
@@ -697,6 +697,24 @@ class vector_downward {
     }
   }
 
+  void reset(size_t initial_size = 1024, Allocator *allocator = nullptr,
+             bool own_allocator = false) {
```

Comment:
Why not simplify this method to assume we keep the same allocator, since it is only used from GRPC code at the moment? In fact, it is called with the exact same parameters than on construction, so I don't see why we're reallocating the buffer.